File: testbed/test/cruise/test/ReflexiveOptionalOneTest.java
package cruise.test;

import org.junit.*;

public class ReflexiveOptionalOneTest
{

  @Test
  public void SetSuperMentor()
  {
    MentorH m = new MentorH("m1");
    MentorH m2 = new MentorH("m2");
    
    m.setSuperMentor(m2);
    Assert.assertEquals(m, m2.getSuperMentor());
    Assert.assertEquals(m2, m.getSuperMentor());
  }
  
}

File: testbed_ruby/test/associations/reflexive_optional_one_test.rb
require 'test/unit'
require './src-gen-umple/mentor_h'

module CruiseAssociations
class ReflexiveOptionalOneTest < Test::Unit::TestCase

  def test_SetSuperMentor
    m = MentorH.new("m1")
    m2 = MentorH.new("m2")
    
    m.set_superMentor(m2)
    assert_equal(m, m2.get_superMentor)
    assert_equal(m2, m.get_superMentor)
  end
  
end
end